agent/exec/container/executor.go

```diff
@@ -29,6 +30,7 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 
 	var plugins []api.PluginDescription
 	addPlugins := func(typ string, names []string) {
+		sort.Strings(names)
```

Comment:
This sorts on every addition. Do this before setting the field.

Needs to sort by type, as well.

```
type sortedPlugins []PluginDescription

func (sp sortedPlugins) Len() int { return len(sp) }
func (sp sortedPlugins) Swap(i, j int) { sp[i], sp[j] = sp[j], s[i] }
func (sp sortedPlugins) Less(i, j int) bool {
  if sp[i].Type != sp[j].Type {
    return sp[i].Type < sp[j].Type
  }

  return sp[i].Name < sp[j].Name
}
```

Then, before setting the field:

```
sort.Sort(sortedPlugins(plugins))
```